From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nuutti <snuutti@protonmail.com>
Date: Thu, 5 Mar 2026 23:59:37 +0200
Subject: [PATCH] Fix WWWRequest


diff --git a/Assets/Scripts/Assembly-CSharp/WWWRequest.cs b/Assets/Scripts/Assembly-CSharp/WWWRequest.cs
index 33ae90fe..c4836a83 100644
--- a/Assets/Scripts/Assembly-CSharp/WWWRequest.cs
+++ b/Assets/Scripts/Assembly-CSharp/WWWRequest.cs
@@ -4,2 +4,3 @@ using System.Diagnostics;
 using UnityEngine;
+using UnityEngine.Networking;
 
@@ -23,3 +24,3 @@ public class WWWRequest
 
-	private WWW request;
+	private UnityWebRequest request;
 
@@ -95,3 +96,3 @@ public class WWWRequest
 		{
-			return request.bytesDownloaded;
+			return (int) request.downloadedBytes;
 		}
@@ -103,3 +104,3 @@ public class WWWRequest
 		{
-			return (request == null) ? 0f : request.progress;
+			return (request == null) ? 0f : request.downloadProgress;
 		}
@@ -113,3 +114,3 @@ public class WWWRequest
 		{
-			return (request == null) ? null : request.assetBundle;
+			return (request == null) ? null : DownloadHandlerAssetBundle.GetContent(request);
 		}
@@ -121,3 +122,3 @@ public class WWWRequest
 		{
-			return (request == null) ? null : request.text;
+			return (request == null || !request.isDone) ? null : request.downloadHandler.text;
 		}
@@ -155,3 +156,3 @@ public class WWWRequest
 			{
-				request = WWW.LoadFromCacheOrDownload(URI, 0, CRC);
+				request = UnityWebRequestAssetBundle.GetAssetBundle(URI, CRC);
 			}
@@ -159,7 +160,18 @@ public class WWWRequest
 			{
-				request = new WWW(URI);
+				request = new UnityWebRequest(URI);
 			}
+
+			request.SendWebRequest();
 			break;
 		case DataType.Text:
-			request = new WWW(URI, PostData, Headers);
+			if (PostData == null)
+			{
+				request = new UnityWebRequest(URI, "GET");
+			}
+			else
+			{
+				request = new UnityWebRequest(URI, "POST");
+			}
+			request.downloadHandler = new DownloadHandlerBuffer();
+			request.SendWebRequest();
 			break;
@@ -215,3 +227,3 @@ public class WWWRequest
 		float realtimeSinceStartup = Time.realtimeSinceStartup;
-		if (lastProgress == request.progress)
+		if (lastProgress == request.downloadProgress)
 		{
@@ -230,3 +242,3 @@ public class WWWRequest
 		{
-			lastProgress = request.progress;
+			lastProgress = request.downloadProgress;
 			progressTimestamp = realtimeSinceStartup;
