From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nuutti <snuutti@protonmail.com>
Date: Fri, 6 Mar 2026 00:00:03 +0200
Subject: [PATCH] Add GameDebug Log implementation


diff --git a/Assets/Scripts/Assembly-CSharp/GameDebug.cs b/Assets/Scripts/Assembly-CSharp/GameDebug.cs
index 887d0451..6f7f00f6 100644
--- a/Assets/Scripts/Assembly-CSharp/GameDebug.cs
+++ b/Assets/Scripts/Assembly-CSharp/GameDebug.cs
@@ -99,2 +99,20 @@ public static class GameDebug
 	{
+#if UNITY_EDITOR
+		var msg = string.Format(format, args);
+		switch (level)
+		{
+			case LogLevel.Critical:
+			case LogLevel.Error:
+				Debug.LogError(msg, context);
+				break;
+
+			case LogLevel.Warning:
+				Debug.LogWarning(msg, context);
+				break;
+
+			default:
+				Debug.Log(msg, context);
+				break;
+		}
+#endif
 	}
