From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nuutti <snuutti@protonmail.com>
Date: Wed, 25 Mar 2026 23:50:16 +0200
Subject: [PATCH] Build settings


diff --git a/ProjectSettings/ProjectSettings.asset b/ProjectSettings/ProjectSettings.asset
index 1dd82cca..02e48192 100644
--- a/ProjectSettings/ProjectSettings.asset
+++ b/ProjectSettings/ProjectSettings.asset
@@ -94,5 +94,5 @@ PlayerSettings:
   bakeCollisionMeshes: 0
-  forceSingleInstance: 0
+  forceSingleInstance: 1
   useFlipModelSwapchain: 1
-  resizableWindow: 0
+  resizableWindow: 1
   useMacAppStoreValidation: 0
@@ -165,7 +165,7 @@ PlayerSettings:
   applicationIdentifier:
-    Android: com.nexonm.tffl
-    Standalone: unity.Particle City.Frontline
-    VisionOS: com.nexonm.tffl
-    iPhone: com.nexonm.tffl
-    tvOS: com.nexonm.tffl
+    Android: com.ParticleCity.Frontline
+    Standalone: com.Particle-City.Frontline
+    VisionOS: com.ParticleCity.Frontline
+    iPhone: com.ParticleCity.Frontline
+    tvOS: com.ParticleCity.Frontline
   buildNumber:
@@ -176,3 +176,3 @@ PlayerSettings:
   overrideDefaultApplicationIdentifier: 0
-  AndroidBundleVersionCode: 0
+  AndroidBundleVersionCode: 1
   AndroidMinSdkVersion: 22
@@ -269,3 +269,3 @@ PlayerSettings:
   useCustomProguardFile: 0
-  AndroidTargetArchitectures: 1
+  AndroidTargetArchitectures: 3
   AndroidTargetDevices: 0
@@ -643,3 +643,5 @@ PlayerSettings:
   platformArchitecture: {}
-  scriptingBackend: {}
+  scriptingBackend:
+    Android: 1
+    Standalone: 1
   il2cppCompilerConfiguration: {}
@@ -745,3 +747,3 @@ PlayerSettings:
   hmiCpuConfiguration: 
-  apiCompatibilityLevel: 3
+  apiCompatibilityLevel: 6
   activeInputHandler: 0
@@ -758,2 +760,2 @@ PlayerSettings:
   virtualTexturingSupportEnabled: 0
-  insecureHttpOption: 0
+  insecureHttpOption: 2
diff --git a/ProjectSettings/QualitySettings.asset b/ProjectSettings/QualitySettings.asset
index cfec5e87..a1f21908 100644
--- a/ProjectSettings/QualitySettings.asset
+++ b/ProjectSettings/QualitySettings.asset
@@ -6,3 +6,3 @@ QualitySettings:
   serializedVersion: 5
-  m_CurrentQuality: 0
+  m_CurrentQuality: 4
   m_QualitySettings:
@@ -261,3 +261,4 @@ QualitySettings:
     PSP2: 2
-    Standalone: 5
+    Server: 0
+    Standalone: 4
     Tizen: 2
