From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nuutti <snuutti@protonmail.com>
Date: Thu, 5 Mar 2026 23:53:53 +0200
Subject: [PATCH] Remove platform specific code


diff --git a/Assets/Scripts/Assembly-CSharp/AssetManager.cs b/Assets/Scripts/Assembly-CSharp/AssetManager.cs
index 076b6022..0afa5573 100644
--- a/Assets/Scripts/Assembly-CSharp/AssetManager.cs
+++ b/Assets/Scripts/Assembly-CSharp/AssetManager.cs
@@ -121,2 +121,3 @@ public class AssetManager : Singleton<AssetManager>
 		MaxActiveSceneRequests = 1;
+#if !UNITY_WEBGL
 		Caching.maximumAvailableDiskSpace = MAX_CACHE_SPACE;
@@ -126,2 +127,3 @@ public class AssetManager : Singleton<AssetManager>
 		}
+#endif
 		bundleURIs = new List<string>();
diff --git a/Assets/Scripts/Assembly-CSharp/CCGHelpshift.cs b/Assets/Scripts/Assembly-CSharp/CCGHelpshift.cs
index b302947b..fb6395bb 100644
--- a/Assets/Scripts/Assembly-CSharp/CCGHelpshift.cs
+++ b/Assets/Scripts/Assembly-CSharp/CCGHelpshift.cs
@@ -18,8 +18,2 @@ public class CCGHelpshift : MonoBehaviour
 
-	private void Start()
-	{
-		Object.DontDestroyOnLoad(base.gameObject);
-		SetupHelpshift();
-	}
-
 	public void SetupHelpshift()
diff --git a/Assets/Scripts/Assembly-CSharp/GameShell.cs b/Assets/Scripts/Assembly-CSharp/GameShell.cs
index df65a07f..1e65d23e 100644
--- a/Assets/Scripts/Assembly-CSharp/GameShell.cs
+++ b/Assets/Scripts/Assembly-CSharp/GameShell.cs
@@ -147,3 +147,2 @@ public class GameShell : SingletonMonoBehaviour<GameShell>, INPListenerType, INP
 		}
-		SetupAdjust();
 		screenWidth = Screen.width;
@@ -349,2 +348,3 @@ public class GameShell : SingletonMonoBehaviour<GameShell>, INPListenerType, INP
 		case RuntimePlatform.LinuxPlayer:
+		case RuntimePlatform.LinuxEditor:
 		case RuntimePlatform.WebGLPlayer:
@@ -711,5 +711,2 @@ public class GameShell : SingletonMonoBehaviour<GameShell>, INPListenerType, INP
 	{
-		string text = Application.identifier.Substring(Application.identifier.LastIndexOf('.') + 1);
-		string text2 = text + "." + Application.version;
-		CrashReporting.Init("fd1f4b60-2c24-415e-b8d8-dd633c752c22", text2, userID);
 	}
diff --git a/Assets/Scripts/Assembly-CSharp/PvE/PvEMissionsTemplateParser.cs b/Assets/Scripts/Assembly-CSharp/PvE/PvEMissionsTemplateParser.cs
index 3c92b0da..bc7e272a 100644
--- a/Assets/Scripts/Assembly-CSharp/PvE/PvEMissionsTemplateParser.cs
+++ b/Assets/Scripts/Assembly-CSharp/PvE/PvEMissionsTemplateParser.cs
@@ -93,4 +93,6 @@ namespace PvE
 					Dictionary<string, object> pages = fastJSON.JSON.Parse(text2) as Dictionary<string, object>;
+#if !UNITY_WEBGL
 					PlayerPrefs.SetString("PvETemplateVersion", text);
 					PlayerPrefs.SetString("PvETemplateData", text2);
+#endif
 					GameDebug.Info("PvE:Template:Local mission data bad! Using data the server sent back.");
@@ -116,3 +118,7 @@ namespace PvE
 			bool finished = false;
+#if !UNITY_WEBGL
 			string hash = PlayerPrefs.GetString("PvETemplateVersion", string.Empty);
+#else
+			string hash = string.Empty;
+#endif
 			OperationGetMissionData httpOperation = new OperationGetMissionData(hash);
diff --git a/Assets/Scripts/Assembly-CSharp/SimpleDiskUtils/DiskUtils.cs b/Assets/Scripts/Assembly-CSharp/SimpleDiskUtils/DiskUtils.cs
index b20d3baf..fbbd256e 100644
--- a/Assets/Scripts/Assembly-CSharp/SimpleDiskUtils/DiskUtils.cs
+++ b/Assets/Scripts/Assembly-CSharp/SimpleDiskUtils/DiskUtils.cs
@@ -11,4 +11,3 @@ namespace SimpleDiskUtils
 		{
-			AndroidJavaClass androidJavaClass = new AndroidJavaClass("com.dikra.diskutils.DiskUtils");
-			return androidJavaClass.CallStatic<int>("availableSpace", new object[1] { isExternalStorage });
+			return int.MaxValue;
 		}
