From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nuutti <snuutti@protonmail.com>
Date: Fri, 6 Mar 2026 02:48:32 +0200
Subject: [PATCH] Remove Nexon analytics


diff --git a/Assets/Scripts/Assembly-CSharp/FUE_Script.cs b/Assets/Scripts/Assembly-CSharp/FUE_Script.cs
index 6c5a5f98..a22e2bb5 100644
--- a/Assets/Scripts/Assembly-CSharp/FUE_Script.cs
+++ b/Assets/Scripts/Assembly-CSharp/FUE_Script.cs
@@ -213,3 +213,2 @@ public class FUE_Script : BaseGameScript
 		}
-		NxSignal.reportCustomEvent("funnel_f", new Dictionary<string, object> { { "funnel_step", "Turn 1: Player Deploys Susan Raymos" } });
 		handFX.SetActive(false);
@@ -243,3 +242,2 @@ public class FUE_Script : BaseGameScript
 	{
-		NxSignal.reportCustomEvent("funnel_f", new Dictionary<string, object> { { "funnel_step", "Turn 1: Player ends turn 1" } });
 		yield return new WaitForSeconds(1f);
@@ -297,3 +295,2 @@ public class FUE_Script : BaseGameScript
 		EndAITurn();
-		NxSignal.reportCustomEvent("funnel_f", new Dictionary<string, object> { { "funnel_step", "AI Turn 1: AI completes turn 1" } });
 	}
@@ -313,3 +310,2 @@ public class FUE_Script : BaseGameScript
 		}
-		NxSignal.reportCustomEvent("funnel_f", new Dictionary<string, object> { { "funnel_step", "Turn 2: Deploy Jasmine Kane" } });
 		Object.Destroy(commandFX);
@@ -348,3 +344,2 @@ public class FUE_Script : BaseGameScript
 		gameUI.CloseTalkBubble();
-		NxSignal.reportCustomEvent("funnel_f", new Dictionary<string, object> { { "funnel_step", "Turn 2: Move Susan Raymos" } });
 		StartStep(PlayerEndTurn, AITurn2);
@@ -354,3 +349,2 @@ public class FUE_Script : BaseGameScript
 	{
-		NxSignal.reportCustomEvent("funnel_f", new Dictionary<string, object> { { "funnel_step", "Turn 2: Player ends turn 2" } });
 		yield return new WaitForSeconds(1f);
@@ -399,3 +393,2 @@ public class FUE_Script : BaseGameScript
 		StartStep(WaitForCardDraw, PlayerTurn3Step1);
-		NxSignal.reportCustomEvent("funnel_f", new Dictionary<string, object> { { "funnel_step", "AI Turn 2: AI completes turn 2" } });
 	}
@@ -427,3 +420,2 @@ public class FUE_Script : BaseGameScript
 		}
-		NxSignal.reportCustomEvent("funnel_f", new Dictionary<string, object> { { "funnel_step", "Turn 3: Deploy Field HQ2" } });
 		StartStep(PlayerTurn3Step4);
@@ -448,3 +440,2 @@ public class FUE_Script : BaseGameScript
 		gameUI.CloseTalkBubble();
-		NxSignal.reportCustomEvent("funnel_f", new Dictionary<string, object> { { "funnel_step", "Turn 3: Move Jasmine Kane" } });
 		yield return new WaitForSeconds(1f);
@@ -463,3 +454,2 @@ public class FUE_Script : BaseGameScript
 		gameUI.CloseTalkBubble();
-		NxSignal.reportCustomEvent("funnel_f", new Dictionary<string, object> { { "funnel_step", "Turn 3: Attack with Susan Raymos" } });
 		yield return new WaitForSeconds(0.1f);
@@ -483,3 +473,2 @@ public class FUE_Script : BaseGameScript
 		}
-		NxSignal.reportCustomEvent("funnel_f", new Dictionary<string, object> { { "funnel_step", "Turn 3: Deploy support card" } });
 		StartStep(PlayerEndTurn, AITurn3);
@@ -489,3 +478,2 @@ public class FUE_Script : BaseGameScript
 	{
-		NxSignal.reportCustomEvent("funnel_f", new Dictionary<string, object> { { "funnel_step", "Turn 3: Player ends turn 3" } });
 		yield return new WaitForSeconds(1f);
@@ -524,3 +512,2 @@ public class FUE_Script : BaseGameScript
 		EndAITurn();
-		NxSignal.reportCustomEvent("funnel_f", new Dictionary<string, object> { { "funnel_step", "AI Turn 3: AI completes turn 3" } });
 	}
@@ -544,3 +531,2 @@ public class FUE_Script : BaseGameScript
 		}
-		NxSignal.reportCustomEvent("funnel_f", new Dictionary<string, object> { { "funnel_step", "Turn 4: Deploy Atlas" } });
 		StartStep(PlayerTurn4Step2);
@@ -566,3 +552,2 @@ public class FUE_Script : BaseGameScript
 		}
-		NxSignal.reportCustomEvent("funnel_f", new Dictionary<string, object> { { "funnel_step", "Turn 4: Attack with Atlass" } });
 		StartStep(PlayerTurn4Step3);
@@ -583,3 +568,2 @@ public class FUE_Script : BaseGameScript
 		}
-		NxSignal.reportCustomEvent("funnel_f", new Dictionary<string, object> { { "funnel_step", "Turn 4: Attack with Susan Raymos" } });
 		StartStep(PlayerEndTurn, AITurn4);
@@ -599,3 +583,2 @@ public class FUE_Script : BaseGameScript
 	{
-		NxSignal.reportCustomEvent("funnel_f", new Dictionary<string, object> { { "funnel_step", "Turn 4: Player ends turn 4" } });
 		yield return new WaitForSeconds(1f);
@@ -644,3 +627,2 @@ public class FUE_Script : BaseGameScript
 		EndAITurn();
-		NxSignal.reportCustomEvent("funnel_f", new Dictionary<string, object> { { "funnel_step", "AI Turn 4: AI completes turn 4" } });
 	}
@@ -662,3 +644,2 @@ public class FUE_Script : BaseGameScript
 		gameUI.CloseTalkBubble();
-		NxSignal.reportCustomEvent("funnel_f", new Dictionary<string, object> { { "funnel_step", "Turn 5: Deploy Barrage" } });
 		yield return new WaitForSeconds(3.5f);
@@ -685,3 +666,2 @@ public class FUE_Script : BaseGameScript
 		}
-		NxSignal.reportCustomEvent("funnel_f", new Dictionary<string, object> { { "funnel_step", "Turn 5: Attack with Susan Raymos" } });
 		yield return new WaitForSeconds(0.1f);
@@ -698,3 +678,2 @@ public class FUE_Script : BaseGameScript
 		}
-		NxSignal.reportCustomEvent("funnel_f", new Dictionary<string, object> { { "funnel_step", "Turn 5: Attack with Atlas" } });
 	}
@@ -753,3 +732,2 @@ public class FUE_Script : BaseGameScript
 	{
-		NxSignal.reportCustomEvent("funnel_f", new Dictionary<string, object> { { "funnel_step", "nx_ftue_end" } });
 		TutorialManager.SetCompletedTutorial(Tutorials.FirstUserExperience);
diff --git a/Assets/Scripts/Assembly-CSharp/MAPManager.cs b/Assets/Scripts/Assembly-CSharp/MAPManager.cs
index fe105732..ea02e12a 100644
--- a/Assets/Scripts/Assembly-CSharp/MAPManager.cs
+++ b/Assets/Scripts/Assembly-CSharp/MAPManager.cs
@@ -20,9 +20,2 @@ public class MAPManager : Singleton<MAPManager>
 			dictionary.Add("developer_player_id", AccountManager.profile.userId.ToString());
-			NxSignal.setFieldDefaultValues(dictionary);
-			NxSignal.reportCustomEvent("launch_f", new Dictionary<string, object> { { "launch_type", "login" } });
-			NxSignal.reportCustomEvent("user_f", new Dictionary<string, object> { 
-			{
-				"username",
-				AccountManager.profile.name
-			} });
 		}
@@ -55,3 +48,2 @@ public class MAPManager : Singleton<MAPManager>
 			}
-			NxSignal.reportRevenueEvent(productId, res.product.LocalizedPrice, res.product.CurrencyCode, res.TransactionId, storeId, "currency", new Dictionary<string, object>());
 		}
@@ -187,3 +179,2 @@ public class MAPManager : Singleton<MAPManager>
 			dictionary.Add("interaction_turn", Convert.ToInt16(game.gameState.currentRound));
-			NxSignal.reportCustomEvent("battle_interaction_f", dictionary);
 		}
@@ -270,3 +261,2 @@ public class MAPManager : Singleton<MAPManager>
 			dictionary.Add("result_duration", durationSeconds);
-			NxSignal.reportCustomEvent("battle_result_f", dictionary);
 		}
@@ -298,3 +288,2 @@ public class MAPManager : Singleton<MAPManager>
 			dictionary.Add("target", product.ProductID);
-			NxSignal.reportCustomEvent("econ_sink_f", dictionary);
 		}
@@ -326,3 +315,2 @@ public class MAPManager : Singleton<MAPManager>
 				dictionary.Add("action", action);
-				NxSignal.reportCustomEvent("econ_source_f", dictionary);
 			}
@@ -342,3 +330,2 @@ public class MAPManager : Singleton<MAPManager>
 			dictionary.Add("isStart ", begin);
-			NxSignal.reportCustomEvent("activity_f", dictionary);
 		}
@@ -352,3 +339,2 @@ public class MAPManager : Singleton<MAPManager>
 			dictionary.Add("error_f", error);
-			NxSignal.reportCustomEvent("error_f", dictionary);
 		}
@@ -360,3 +346,2 @@ public class MAPManager : Singleton<MAPManager>
 		dictionary.Add("error_f", "not_enough_space");
-		NxSignal.reportCustomEvent("error_f", dictionary);
 	}
diff --git a/Assets/Scripts/Assembly-CSharp/MainUI.cs b/Assets/Scripts/Assembly-CSharp/MainUI.cs
index 4fc8e2ae..e95c467b 100644
--- a/Assets/Scripts/Assembly-CSharp/MainUI.cs
+++ b/Assets/Scripts/Assembly-CSharp/MainUI.cs
@@ -452,3 +452,2 @@ public class MainUI : BaseUI
 			Debug.Log("SHOW FIRST USER...");
-			NxSignal.reportCustomEvent("screenview_f", new Dictionary<string, object> { { "screen", "FirstUser" } });
 			firstUserUI.gameObject.SetActive(true);
@@ -458,3 +457,2 @@ public class MainUI : BaseUI
 			loginDialog.transform.parent.gameObject.SetActive(true);
-			NxSignal.reportCustomEvent("screenview_f", new Dictionary<string, object> { { "screen", "Login" } });
 			break;
@@ -462,3 +460,2 @@ public class MainUI : BaseUI
 			ConnectingUI.SetActive(true);
-			NxSignal.reportCustomEvent("screenview_f", new Dictionary<string, object> { { "screen", "Connecting" } });
 			_connectingTimerHandle = Singleton<TimerManager>.Instance.AddTimer(_connectingTimeoutTime, OnConnectingTimeout, null, string.Empty);
@@ -468,3 +465,2 @@ public class MainUI : BaseUI
 			currentProcess = Process.waitForRulesetToLoad;
-			NxSignal.reportCustomEvent("screenview_f", new Dictionary<string, object> { { "screen", "MenuScreen" } });
 			break;
@@ -478,3 +474,2 @@ public class MainUI : BaseUI
 			Singleton<UIStateManager>.Instance.SetState(UIState.State.EnteringGame);
-			NxSignal.reportCustomEvent("screenview_f", new Dictionary<string, object> { { "screen", "MatchMaking" } });
 			break;
@@ -485,3 +480,2 @@ public class MainUI : BaseUI
 			backButton.gameObject.SetActive(true);
-			NxSignal.reportCustomEvent("screenview_f", new Dictionary<string, object> { { "screen", "MissionLeaderboard" } });
 			break;
@@ -491,3 +485,2 @@ public class MainUI : BaseUI
 			backButton.gameObject.SetActive(true);
-			NxSignal.reportCustomEvent("screenview_f", new Dictionary<string, object> { { "screen", "GuildLeaderboard" } });
 			break;
@@ -497,3 +490,2 @@ public class MainUI : BaseUI
 			backButton.gameObject.SetActive(true);
-			NxSignal.reportCustomEvent("screenview_f", new Dictionary<string, object> { { "screen", "EventLeaderboard" } });
 			break;
@@ -504,3 +496,2 @@ public class MainUI : BaseUI
 				backButton.gameObject.SetActive(true);
-				NxSignal.reportCustomEvent("screenview_f", new Dictionary<string, object> { { "screen", "News" } });
 			}
@@ -510,3 +501,2 @@ public class MainUI : BaseUI
 			backButton.gameObject.SetActive(true);
-			NxSignal.reportCustomEvent("screenview_f", new Dictionary<string, object> { { "screen", "ChatUI" } });
 			break;
@@ -514,3 +504,2 @@ public class MainUI : BaseUI
 			Singleton<UIStateManager>.Instance.SetState(UIState.State.Store);
-			NxSignal.reportCustomEvent("screenview_f", new Dictionary<string, object> { { "screen", "StoryUI" } });
 			break;
diff --git a/Assets/Scripts/Assembly-CSharp/PvE/PvEClientManager.cs b/Assets/Scripts/Assembly-CSharp/PvE/PvEClientManager.cs
index 84f7e868..e6a6ff63 100644
--- a/Assets/Scripts/Assembly-CSharp/PvE/PvEClientManager.cs
+++ b/Assets/Scripts/Assembly-CSharp/PvE/PvEClientManager.cs
@@ -587,7 +587,2 @@ namespace PvE
 		{
-			NxSignal.reportCustomEvent("entered_missions_f", new Dictionary<string, object> { 
-			{
-				"player_id",
-				AccountManager.profile.userId.ToString()
-			} });
 			if (Me != null)
@@ -607,7 +602,2 @@ namespace PvE
 		{
-			NxSignal.reportCustomEvent("exited_missions_f", new Dictionary<string, object> { 
-			{
-				"player_id",
-				AccountManager.profile.userId.ToString()
-			} });
 		}
@@ -865,32 +855,2 @@ namespace PvE
 					string value = ((eRegion)missionStageStatus.region).ToString() + "-" + ((Faction)missionStageStatus.faction).ToString() + "-" + missionStageStatus.missionid.ToString("D4");
-					NxSignal.reportCustomEvent("mission_completed_f", new Dictionary<string, object>
-					{
-						{
-							"player_id",
-							AccountManager.profile.userId.ToString()
-						},
-						{ "mission_id", value },
-						{ "start_time", missionStageStatus.missionstagestart },
-						{ "finish_time", Now },
-						{
-							"card0_used",
-							missionStageStatus.card0instanceid != 0
-						},
-						{
-							"card1_used",
-							missionStageStatus.card1instanceid != 0
-						},
-						{
-							"card2_used",
-							missionStageStatus.card2instanceid != 0
-						},
-						{
-							"synergy0_used",
-							missionStageStatus.synergy0reward0 != null || missionStageStatus.synergy0Effect != null
-						},
-						{
-							"synergy1_used",
-							missionStageStatus.synergy1reward0 != null || missionStageStatus.synergy1Effect != null
-						}
-					});
 				}
diff --git a/Assets/Scripts/Assembly-CSharp/UIManager.cs b/Assets/Scripts/Assembly-CSharp/UIManager.cs
index 3936bdc0..48165b04 100644
--- a/Assets/Scripts/Assembly-CSharp/UIManager.cs
+++ b/Assets/Scripts/Assembly-CSharp/UIManager.cs
@@ -148,7 +148,2 @@ public class UIManager : MonoBehaviour
 			{
-				NxSignal.reportCustomEvent("screenview_f", new Dictionary<string, object> { 
-				{
-					"screen",
-					newMode.ToString()
-				} });
 				currentUI = UnityEngine.Object.Instantiate(gameObject);
