From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nuutti <snuutti@protonmail.com>
Date: Thu, 5 Mar 2026 23:53:53 +0200
Subject: [PATCH] Remove platform specific code


diff --git a/Assets/Scripts/Assembly-CSharp/CCGHelpshift.cs b/Assets/Scripts/Assembly-CSharp/CCGHelpshift.cs
index b302947b..fb6395bb 100644
--- a/Assets/Scripts/Assembly-CSharp/CCGHelpshift.cs
+++ b/Assets/Scripts/Assembly-CSharp/CCGHelpshift.cs
@@ -18,8 +18,2 @@ public class CCGHelpshift : MonoBehaviour
 
-	private void Start()
-	{
-		Object.DontDestroyOnLoad(base.gameObject);
-		SetupHelpshift();
-	}
-
 	public void SetupHelpshift()
diff --git a/Assets/Scripts/Assembly-CSharp/GameShell.cs b/Assets/Scripts/Assembly-CSharp/GameShell.cs
index df65a07f..1e65d23e 100644
--- a/Assets/Scripts/Assembly-CSharp/GameShell.cs
+++ b/Assets/Scripts/Assembly-CSharp/GameShell.cs
@@ -147,3 +147,2 @@ public class GameShell : SingletonMonoBehaviour<GameShell>, INPListenerType, INP
 		}
-		SetupAdjust();
 		screenWidth = Screen.width;
@@ -349,2 +348,3 @@ public class GameShell : SingletonMonoBehaviour<GameShell>, INPListenerType, INP
 		case RuntimePlatform.LinuxPlayer:
+		case RuntimePlatform.LinuxEditor:
 		case RuntimePlatform.WebGLPlayer:
@@ -711,5 +711,2 @@ public class GameShell : SingletonMonoBehaviour<GameShell>, INPListenerType, INP
 	{
-		string text = Application.identifier.Substring(Application.identifier.LastIndexOf('.') + 1);
-		string text2 = text + "." + Application.version;
-		CrashReporting.Init("fd1f4b60-2c24-415e-b8d8-dd633c752c22", text2, userID);
 	}
diff --git a/Assets/Scripts/Assembly-CSharp/MainUI.cs b/Assets/Scripts/Assembly-CSharp/MainUI.cs
index 9e18c400..652e3c25 100644
--- a/Assets/Scripts/Assembly-CSharp/MainUI.cs
+++ b/Assets/Scripts/Assembly-CSharp/MainUI.cs
@@ -206,2 +206,3 @@ public class MainUI : BaseUI
 	{
+#if UNITY_ANDROID
 		int num = Screen.height;
@@ -214,2 +215,3 @@ public class MainUI : BaseUI
 		Screen.SetResolution(num2, num, true);
+#endif
 		base.Start();
diff --git a/Assets/Scripts/Assembly-CSharp/SimpleDiskUtils/DiskUtils.cs b/Assets/Scripts/Assembly-CSharp/SimpleDiskUtils/DiskUtils.cs
index b20d3baf..fbbd256e 100644
--- a/Assets/Scripts/Assembly-CSharp/SimpleDiskUtils/DiskUtils.cs
+++ b/Assets/Scripts/Assembly-CSharp/SimpleDiskUtils/DiskUtils.cs
@@ -11,4 +11,3 @@ namespace SimpleDiskUtils
 		{
-			AndroidJavaClass androidJavaClass = new AndroidJavaClass("com.dikra.diskutils.DiskUtils");
-			return androidJavaClass.CallStatic<int>("availableSpace", new object[1] { isExternalStorage });
+			return int.MaxValue;
 		}
