From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nuutti <snuutti@protonmail.com>
Date: Thu, 5 Mar 2026 23:54:26 +0200
Subject: [PATCH] TOY

TODO: We need a cross platform C# TOY client implementation.

diff --git a/Assets/Scripts/Assembly-CSharp/GameShell.cs b/Assets/Scripts/Assembly-CSharp/GameShell.cs
index 8272ce18..3e1f5634 100644
--- a/Assets/Scripts/Assembly-CSharp/GameShell.cs
+++ b/Assets/Scripts/Assembly-CSharp/GameShell.cs
@@ -849,3 +849,8 @@ public class GameShell : SingletonMonoBehaviour<GameShell>, INPListenerType, INP
 		{
-			account.EnterToy(this);
+			//account.EnterToy(this);
+			OnResult(new NPResult
+			{
+				requestTag = NPRequestTypeTag.NPRequestTypeGetUserInfo,
+				resultJson = JSONNode.Parse("{\"errorCode\":0,\"result\":{\"npsn\":\"\"12530000000000005\"\",\"npToken\":\"eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiIiLCJpYXQiOjE3NzI3NDI5NTcsImV4cCI6MTc3NTQyMTM1NywiYXVkIjoiIiwic3ViIjoiIiwiVXNlcklkIjoiNSIsIlNlcnZpY2VJZCI6IjEyNTMifQ.jDZOVRNkrmHmJHWWQXOYLmcegcYXgZUSv1iyyw9Iz-w\"}}")
+			});
 		}
