From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nuutti <snuutti@protonmail.com>
Date: Tue, 21 Apr 2026 22:28:14 +0300
Subject: [PATCH] Replace bad version check with Semver


diff --git a/Assets/Scripts/Assembly-CSharp/GameShell.cs b/Assets/Scripts/Assembly-CSharp/GameShell.cs
index 779713d6..be5a302f 100644
--- a/Assets/Scripts/Assembly-CSharp/GameShell.cs
+++ b/Assets/Scripts/Assembly-CSharp/GameShell.cs
@@ -9,2 +9,3 @@ using UnityEngine.CrashLog;
 using fastJSON;
+using Semver;
 
@@ -430,44 +431,11 @@ public class GameShell : SingletonMonoBehaviour<GameShell>, INPListenerType, INP
 			}
-			bool flag = false;
-			char[] separator = new char[1] { '.' };
-			char[] trimChars = new char[1] { '0' };
+			bool flag = true;
 			string text = (string)dictionary["minClientVersion"];
-			string[] array = text.Split(separator);
-			string[] array2 = version.Split(separator);
-			if (array2.Length == 0 || array.Length == 0)
+			if (SemVersion.TryParse(text, out var minVersion))
 			{
-				GameDebug.Error("Invalid version formats - Version {0} MinVersion {1}", version, text);
-				flag = true;
-			}
-			else if (array2.Length != array.Length)
-			{
-				GameDebug.Warning("Differing version formats - Version {0} MinVersion {1}", version, text);
-			}
-			int num = Mathf.Min(array2.Length, array.Length);
-			for (int i = 0; i < num; i++)
-			{
-				string text2 = array2[i];
-				if (text2.Length > 1)
+				var currentVersion = SemVersion.Parse(Application.version);
+				if (minVersion.ComparePrecedenceTo(currentVersion) <= 0)
 				{
-					text2 = text2.TrimStart(trimChars);
+					flag = false;
 				}
-				int result = 0;
-				string text3 = array[i];
-				if (text3.Length > 1)
-				{
-					text3 = text3.TrimStart(trimChars);
-				}
-				int result2 = 0;
-				if (int.TryParse(text2, out result) && int.TryParse(text3, out result2))
-				{
-					if (result < result2)
-					{
-						flag = true;
-						break;
-					}
-					continue;
-				}
-				GameDebug.Error("Invalid Version value - Version \"{0}\" {1} MinVersion \"{2}\" {3}", array2[i], version, array[i], text);
-				flag = true;
-				break;
 			}
@@ -475,3 +443,3 @@ public class GameShell : SingletonMonoBehaviour<GameShell>, INPListenerType, INP
 			{
-				ShowUpdateAppPopup();
+				ShowUpdateAppPopup(text);
 				return;
@@ -895,5 +863,5 @@ public class GameShell : SingletonMonoBehaviour<GameShell>, INPListenerType, INP
 
-	public void ShowUpdateAppPopup()
+	public void ShowUpdateAppPopup(string minVersion = "")
 	{
-		string errorText = SingletonMonoBehaviour<LocalizationManager>.Instance.GetErrorText(ErrorModalType.NeedUpdate);
+		var errorText = $"Your game is out of date (version {Application.version})! Please update to the latest version{(string.IsNullOrEmpty(minVersion) ? "" : $" {minVersion}")} to continue playing.";
 		PopUpManager.ShowOk("Error", errorText, OpenUpdatePage);
diff --git a/Packages/nuget-packages/packages.config b/Packages/nuget-packages/packages.config
index c0ab7ed4..fb6e653e 100644
--- a/Packages/nuget-packages/packages.config
+++ b/Packages/nuget-packages/packages.config
@@ -3,3 +3,5 @@
   <package id="DnsClient" version="1.7.0" />
+  <package id="Microsoft.Extensions.Primitives" version="3.0.3" />
   <package id="Microsoft.Win32.Registry" version="5.0.0" />
+  <package id="Semver" version="3.0.0" manuallyInstalled="true" />
   <package id="System.Reactive" version="6.0.1" />
