From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nuutti <snuutti@protonmail.com>
Date: Fri, 6 Mar 2026 00:00:44 +0200
Subject: [PATCH] Fix store purchases


diff --git a/Assets/Scripts/Assembly-CSharp/StoreManager.cs b/Assets/Scripts/Assembly-CSharp/StoreManager.cs
index 424e7452..34eb2e75 100644
--- a/Assets/Scripts/Assembly-CSharp/StoreManager.cs
+++ b/Assets/Scripts/Assembly-CSharp/StoreManager.cs
@@ -55,3 +55,3 @@ public class StoreManager : SingletonMonoBehaviour<StoreManager>
 	{
-		if (IsPurchasePending || !IsAvailable)
+		if (IsPurchasePending || (!IsAvailable && currency == StoreCurrencyType.Real))
 		{
diff --git a/Assets/Scripts/Assembly-CSharp/StoreUI.cs b/Assets/Scripts/Assembly-CSharp/StoreUI.cs
index 992fd5e7..1213f7c2 100644
--- a/Assets/Scripts/Assembly-CSharp/StoreUI.cs
+++ b/Assets/Scripts/Assembly-CSharp/StoreUI.cs
@@ -167,3 +167,3 @@ public class StoreUI : MonoBehaviour
 	{
-		if (SingletonMonoBehaviour<StoreManager>.Instance == null || !SingletonMonoBehaviour<StoreManager>.Instance.IsAvailable || SingletonMonoBehaviour<StoreManager>.Instance.IsPurchasePending)
+		if (SingletonMonoBehaviour<StoreManager>.Instance == null || (!SingletonMonoBehaviour<StoreManager>.Instance.IsAvailable && currency == StoreManager.StoreCurrencyType.Real) || SingletonMonoBehaviour<StoreManager>.Instance.IsPurchasePending)
 		{
