From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nuutti <snuutti@protonmail.com>
Date: Thu, 16 Apr 2026 01:36:33 +0300
Subject: [PATCH] Fix random ArgumentNullException


diff --git a/Assets/Scripts/Assembly-CSharp/FXS_UIFader.cs b/Assets/Scripts/Assembly-CSharp/FXS_UIFader.cs
index 568fc754..9d6c60ec 100644
--- a/Assets/Scripts/Assembly-CSharp/FXS_UIFader.cs
+++ b/Assets/Scripts/Assembly-CSharp/FXS_UIFader.cs
@@ -1 +1,2 @@
+using System;
 using System.Collections.Generic;
@@ -65,4 +66,10 @@ public class FXS_UIFader : MonoBehaviour
 			{
-				Color color3 = textMeshProUGUI.faceColor;
-				textMeshProUGUI.faceColor = new Color(color3.r, color3.g, color3.b, fader);
+				try
+				{
+					Color color3 = textMeshProUGUI.faceColor;
+					textMeshProUGUI.faceColor = new Color(color3.r, color3.g, color3.b, fader);
+				}
+				catch (ArgumentNullException)
+				{
+				}
 			}
