From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nuutti <snuutti@protonmail.com>
Date: Sat, 11 Apr 2026 18:02:07 +0300
Subject: [PATCH] test


diff --git a/Assets/Scripts/Assembly-CSharp/MatrixClient.cs b/Assets/Scripts/Assembly-CSharp/MatrixClient.cs
index 1de3137b..0b79ed27 100644
--- a/Assets/Scripts/Assembly-CSharp/MatrixClient.cs
+++ b/Assets/Scripts/Assembly-CSharp/MatrixClient.cs
@@ -61,8 +61,12 @@ public class MatrixClient : IDisposable, IChatClient
 	{
+		UnityEngine.Debug.Log("CHAT: SetupClient called");
 		if (config == null || string.IsNullOrEmpty(config.UserId) || string.IsNullOrEmpty(config.Password))
 		{
+			UnityEngine.Debug.LogError("CHAT: Invalid configuration");
 			return false;
 		}
+		UnityEngine.Debug.Log("CHAT: Configuration is valid");
 		if (_xmppClient == null)
 		{
+			UnityEngine.Debug.Log("CHAT: Creating XmppClient");
 			_xmppClient = new XmppClient(conf =>
@@ -79,4 +83,10 @@ public class MatrixClient : IDisposable, IChatClient
 				{
+					UnityEngine.Debug.Log("CHAT: OnBind");
 					OnBind();
 				});
+			_xmppClient.XmppXElementReceived
+				.Subscribe(el =>
+				{
+					UnityEngine.Debug.Log("CHAT: Received " + el);
+				});
 			_xmppClient.XmppXElementReceived
@@ -103,3 +113,5 @@ public class MatrixClient : IDisposable, IChatClient
 				});
+			UnityEngine.Debug.Log("CHAT: Connecting...");
 			_xmppClient.ConnectAsync();
+			UnityEngine.Debug.Log("CHAT: ConnectAsync called");
 		}
@@ -139,2 +151,3 @@ public class MatrixClient : IDisposable, IChatClient
 	{
+		UnityEngine.Debug.Log("CHAT: Disconnect called");
 		if (_mucManager != null)
diff --git a/Assets/StreamingAssets/serverAddresses.json b/Assets/StreamingAssets/serverAddresses.json
index 725d86b1..522a282f 100644
--- a/Assets/StreamingAssets/serverAddresses.json
+++ b/Assets/StreamingAssets/serverAddresses.json
@@ -4,9 +4,9 @@
     {
-      "serverAddress": "https://prod-us-east-1-gameserver-lb.tfflinternal.com/",
-      "storeAddress": "https://prod-us-east-1-store-lb.tfflinternal.com/Store/v1",
-      "profileServerAddress": "https://prod-us-east-1-virtu2-lb.tfflinternal.com/",
-      "missionAddress": "https://prod-us-east-1-gameserver-lb.tfflinternal.com/Missions/v1",
-      "dealershipAddress": "https://prod-us-east-1-gameserver-lb.tfflinternal.com/Dealership/v1",
-      "leaderboardAddress": "https://prod-us-east-1-gameserver-lb.tfflinternal.com/",
-      "chatAddress": "prod-us-east-1-chat-lb.tfflinternal.com",
+      "serverAddress": "http://192.168.0.219/",
+      "storeAddress": "http://192.168.0.219/Store/v1",
+      "profileServerAddress": "http://192.168.0.219/",
+      "missionAddress": "http://192.168.0.219/Missions/v1",
+      "dealershipAddress": "http://192.168.0.219/Dealership/v1",
+      "leaderboardAddress": "http://192.168.0.219/",
+      "chatAddress": "192.168.0.219",
       "chatMUCAddress": "conference.prod-us-east-1-chat-lb.tfflinternal.com",
