From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nuutti <snuutti@protonmail.com>
Date: Sat, 28 Mar 2026 04:17:28 +0200
Subject: [PATCH] Mobile target refresh rate


diff --git a/Assets/Scripts/Assembly-CSharp/GameShell.cs b/Assets/Scripts/Assembly-CSharp/GameShell.cs
index 537f29d3..3113c09f 100644
--- a/Assets/Scripts/Assembly-CSharp/GameShell.cs
+++ b/Assets/Scripts/Assembly-CSharp/GameShell.cs
@@ -149,2 +149,6 @@ public class GameShell : SingletonMonoBehaviour<GameShell>, INPListenerType, INP
 		screenHeight = Screen.height;
+#if UNITY_ANDROID || UNITY_IOS
+		// Unity limits FPS to 30 on mobile platforms by default
+		Application.targetFrameRate = (int) Screen.currentResolution.refreshRateRatio.value;
+#endif
 	}
