From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nuutti <snuutti@protonmail.com>
Date: Tue, 24 Mar 2026 04:44:55 +0200
Subject: [PATCH] Enable debug menu


diff --git a/Assets/Resources/DebugMenuButtonTemplate.prefab b/Assets/Resources/DebugMenuButtonTemplate.prefab
index 5e5652ca..35a62dac 100644
--- a/Assets/Resources/DebugMenuButtonTemplate.prefab
+++ b/Assets/Resources/DebugMenuButtonTemplate.prefab
@@ -228,3 +228,3 @@ MonoBehaviour:
     m_MinSize: 8
-    m_MaxSize: 16
+    m_MaxSize: 50
     m_Alignment: 4
diff --git a/Assets/Scenes/GameShell.unity b/Assets/Scenes/GameShell.unity
index 933e977c..bf59ce73 100644
--- a/Assets/Scenes/GameShell.unity
+++ b/Assets/Scenes/GameShell.unity
@@ -32636,2 +32636,203 @@ NavMeshSettings:
   m_NavMeshData: {fileID: 0}
+--- !u!1 &365513272
+GameObject:
+  m_ObjectHideFlags: 0
+  m_CorrespondingSourceObject: {fileID: 0}
+  m_PrefabInstance: {fileID: 0}
+  m_PrefabAsset: {fileID: 0}
+  serializedVersion: 6
+  m_Component:
+  - component: {fileID: 365513274}
+  - component: {fileID: 365513273}
+  - component: {fileID: 365513286}
+  - component: {fileID: 365513285}
+  - component: {fileID: 365513284}
+  - component: {fileID: 365513283}
+  - component: {fileID: 365513282}
+  - component: {fileID: 365513281}
+  - component: {fileID: 365513280}
+  - component: {fileID: 365513279}
+  - component: {fileID: 365513278}
+  - component: {fileID: 365513277}
+  - component: {fileID: 365513276}
+  - component: {fileID: 365513275}
+  m_Layer: 0
+  m_Name: Debug
+  m_TagString: Untagged
+  m_Icon: {fileID: 0}
+  m_NavMeshLayer: 0
+  m_StaticEditorFlags: 0
+  m_IsActive: 1
+--- !u!114 &365513273
+MonoBehaviour:
+  m_ObjectHideFlags: 0
+  m_CorrespondingSourceObject: {fileID: 0}
+  m_PrefabInstance: {fileID: 0}
+  m_PrefabAsset: {fileID: 0}
+  m_GameObject: {fileID: 365513272}
+  m_Enabled: 1
+  m_EditorHideFlags: 0
+  m_Script: {fileID: 11500000, guid: e73b2f213f5abdcc031badd30595fe20, type: 3}
+  m_Name: 
+  m_EditorClassIdentifier: 
+  _parentCanvas: {fileID: 0}
+--- !u!4 &365513274
+Transform:
+  m_ObjectHideFlags: 0
+  m_CorrespondingSourceObject: {fileID: 0}
+  m_PrefabInstance: {fileID: 0}
+  m_PrefabAsset: {fileID: 0}
+  m_GameObject: {fileID: 365513272}
+  serializedVersion: 2
+  m_LocalRotation: {x: 0, y: 0, z: 0, w: 1}
+  m_LocalPosition: {x: 0, y: 0, z: 0}
+  m_LocalScale: {x: 1, y: 1, z: 1}
+  m_ConstrainProportionsScale: 0
+  m_Children: []
+  m_Father: {fileID: 0}
+  m_LocalEulerAnglesHint: {x: 0, y: 0, z: 0}
+--- !u!114 &365513275
+MonoBehaviour:
+  m_ObjectHideFlags: 0
+  m_CorrespondingSourceObject: {fileID: 0}
+  m_PrefabInstance: {fileID: 0}
+  m_PrefabAsset: {fileID: 0}
+  m_GameObject: {fileID: 365513272}
+  m_Enabled: 1
+  m_EditorHideFlags: 0
+  m_Script: {fileID: 11500000, guid: d2d41f2ddbdf7ca5de8f15246252be9e, type: 3}
+  m_Name: 
+  m_EditorClassIdentifier: 
+--- !u!114 &365513276
+MonoBehaviour:
+  m_ObjectHideFlags: 0
+  m_CorrespondingSourceObject: {fileID: 0}
+  m_PrefabInstance: {fileID: 0}
+  m_PrefabAsset: {fileID: 0}
+  m_GameObject: {fileID: 365513272}
+  m_Enabled: 1
+  m_EditorHideFlags: 0
+  m_Script: {fileID: 11500000, guid: c83ca72469c7971d9430b7fa2d2e4c53, type: 3}
+  m_Name: 
+  m_EditorClassIdentifier: 
+--- !u!114 &365513277
+MonoBehaviour:
+  m_ObjectHideFlags: 0
+  m_CorrespondingSourceObject: {fileID: 0}
+  m_PrefabInstance: {fileID: 0}
+  m_PrefabAsset: {fileID: 0}
+  m_GameObject: {fileID: 365513272}
+  m_Enabled: 1
+  m_EditorHideFlags: 0
+  m_Script: {fileID: 11500000, guid: 809bdc3534da9c00cee2fa0c4dbbfcd4, type: 3}
+  m_Name: 
+  m_EditorClassIdentifier: 
+--- !u!114 &365513278
+MonoBehaviour:
+  m_ObjectHideFlags: 0
+  m_CorrespondingSourceObject: {fileID: 0}
+  m_PrefabInstance: {fileID: 0}
+  m_PrefabAsset: {fileID: 0}
+  m_GameObject: {fileID: 365513272}
+  m_Enabled: 1
+  m_EditorHideFlags: 0
+  m_Script: {fileID: 11500000, guid: de3462d8d1bc9adfb5f1e00f6a2660c3, type: 3}
+  m_Name: 
+  m_EditorClassIdentifier: 
+--- !u!114 &365513279
+MonoBehaviour:
+  m_ObjectHideFlags: 0
+  m_CorrespondingSourceObject: {fileID: 0}
+  m_PrefabInstance: {fileID: 0}
+  m_PrefabAsset: {fileID: 0}
+  m_GameObject: {fileID: 365513272}
+  m_Enabled: 1
+  m_EditorHideFlags: 0
+  m_Script: {fileID: 11500000, guid: b54f0f638b530607779f7527a0f19b68, type: 3}
+  m_Name: 
+  m_EditorClassIdentifier: 
+--- !u!114 &365513280
+MonoBehaviour:
+  m_ObjectHideFlags: 0
+  m_CorrespondingSourceObject: {fileID: 0}
+  m_PrefabInstance: {fileID: 0}
+  m_PrefabAsset: {fileID: 0}
+  m_GameObject: {fileID: 365513272}
+  m_Enabled: 1
+  m_EditorHideFlags: 0
+  m_Script: {fileID: 11500000, guid: c839decef3d1935e709ad88cf77108f3, type: 3}
+  m_Name: 
+  m_EditorClassIdentifier: 
+--- !u!114 &365513281
+MonoBehaviour:
+  m_ObjectHideFlags: 0
+  m_CorrespondingSourceObject: {fileID: 0}
+  m_PrefabInstance: {fileID: 0}
+  m_PrefabAsset: {fileID: 0}
+  m_GameObject: {fileID: 365513272}
+  m_Enabled: 1
+  m_EditorHideFlags: 0
+  m_Script: {fileID: 11500000, guid: ca07677518443c07c6f89c3b39a9c638, type: 3}
+  m_Name: 
+  m_EditorClassIdentifier: 
+--- !u!114 &365513282
+MonoBehaviour:
+  m_ObjectHideFlags: 0
+  m_CorrespondingSourceObject: {fileID: 0}
+  m_PrefabInstance: {fileID: 0}
+  m_PrefabAsset: {fileID: 0}
+  m_GameObject: {fileID: 365513272}
+  m_Enabled: 1
+  m_EditorHideFlags: 0
+  m_Script: {fileID: 11500000, guid: 5d8e8b86b8e5d51f477296d8938e2fea, type: 3}
+  m_Name: 
+  m_EditorClassIdentifier: 
+--- !u!114 &365513283
+MonoBehaviour:
+  m_ObjectHideFlags: 0
+  m_CorrespondingSourceObject: {fileID: 0}
+  m_PrefabInstance: {fileID: 0}
+  m_PrefabAsset: {fileID: 0}
+  m_GameObject: {fileID: 365513272}
+  m_Enabled: 1
+  m_EditorHideFlags: 0
+  m_Script: {fileID: 11500000, guid: ed2f62cfe0423ecd987b2d79e1132a78, type: 3}
+  m_Name: 
+  m_EditorClassIdentifier: 
+--- !u!114 &365513284
+MonoBehaviour:
+  m_ObjectHideFlags: 0
+  m_CorrespondingSourceObject: {fileID: 0}
+  m_PrefabInstance: {fileID: 0}
+  m_PrefabAsset: {fileID: 0}
+  m_GameObject: {fileID: 365513272}
+  m_Enabled: 1
+  m_EditorHideFlags: 0
+  m_Script: {fileID: 11500000, guid: a696609a541b4e7b8f0e21f98001d41f, type: 3}
+  m_Name: 
+  m_EditorClassIdentifier: 
+--- !u!114 &365513285
+MonoBehaviour:
+  m_ObjectHideFlags: 0
+  m_CorrespondingSourceObject: {fileID: 0}
+  m_PrefabInstance: {fileID: 0}
+  m_PrefabAsset: {fileID: 0}
+  m_GameObject: {fileID: 365513272}
+  m_Enabled: 1
+  m_EditorHideFlags: 0
+  m_Script: {fileID: 11500000, guid: 0ff7c94f0d19c981e8a1ba20b3aae5be, type: 3}
+  m_Name: 
+  m_EditorClassIdentifier: 
+--- !u!114 &365513286
+MonoBehaviour:
+  m_ObjectHideFlags: 0
+  m_CorrespondingSourceObject: {fileID: 0}
+  m_PrefabInstance: {fileID: 0}
+  m_PrefabAsset: {fileID: 0}
+  m_GameObject: {fileID: 365513272}
+  m_Enabled: 1
+  m_EditorHideFlags: 0
+  m_Script: {fileID: 11500000, guid: 397c41dda309d55cab7f692d8c28ab18, type: 3}
+  m_Name: 
+  m_EditorClassIdentifier: 
 --- !u!1660057539 &9223372036854775807
@@ -32645 +32846,2 @@ SceneRoots:
   - {fileID: 120}
+  - {fileID: 365513274}
diff --git a/Assets/Scripts/Assembly-CSharp/DebugMenuManager.cs b/Assets/Scripts/Assembly-CSharp/DebugMenuManager.cs
index 294c6a96..dc379c4c 100644
--- a/Assets/Scripts/Assembly-CSharp/DebugMenuManager.cs
+++ b/Assets/Scripts/Assembly-CSharp/DebugMenuManager.cs
@@ -4,2 +4,3 @@ using UnityEngine;
 
+[DefaultExecutionOrder(-1)]
 public class DebugMenuManager : SingletonMonoBehaviour<DebugMenuManager>
@@ -265,2 +266,3 @@ public class DebugMenuManager : SingletonMonoBehaviour<DebugMenuManager>
 		base.Awake();
+		DontDestroyOnLoad(this.gameObject);
 		Startup();
@@ -287,2 +289,4 @@ public class DebugMenuManager : SingletonMonoBehaviour<DebugMenuManager>
 		ShouldBeVisible = false;
+		AttachUnderCanvas(GameObject.Find("PopupCanvas"));
+		VisibilityQueryFunc = () => Input.GetKeyDown(KeyCode.F1);
 	}
