From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nuutti <snuutti@protonmail.com>
Date: Sat, 25 Apr 2026 16:45:06 +0300
Subject: [PATCH] Add online player counter to matchmaking screen


diff --git a/Assets/Scripts/Assembly-CSharp/EnteringMatchScreen.cs b/Assets/Scripts/Assembly-CSharp/EnteringMatchScreen.cs
index 3a5019a2..c2c2b6be 100644
--- a/Assets/Scripts/Assembly-CSharp/EnteringMatchScreen.cs
+++ b/Assets/Scripts/Assembly-CSharp/EnteringMatchScreen.cs
@@ -37,2 +37,4 @@ public class EnteringMatchScreen : MonoBehaviour
 
+	public TextMeshProUGUI OnlineText;
+
 	[Header("Background Elements")]
@@ -85,2 +87,6 @@ public class EnteringMatchScreen : MonoBehaviour
 		}
+		if (OnlineText)
+		{
+			OnlineText.gameObject.SetActive(true);
+		}
 	}
@@ -163,2 +169,6 @@ public class EnteringMatchScreen : MonoBehaviour
 			}
+			if (OnlineText.gameObject.activeSelf)
+			{
+				OnlineText.gameObject.SetActive(false);
+			}
 		}
diff --git a/Assets/Scripts/Assembly-CSharp/OnlineCountText.cs b/Assets/Scripts/Assembly-CSharp/OnlineCountText.cs
new file mode 100644
index 00000000..30cec0d9
--- /dev/null
+++ b/Assets/Scripts/Assembly-CSharp/OnlineCountText.cs
@@ -0,0 +1,37 @@
+using System.Collections;
+using Newtonsoft.Json;
+using TMPro;
+using UnityEngine;
+using UnityEngine.Networking;
+
+public class OnlineCountText : MonoBehaviour
+{
+    private TMP_Text _text;
+
+    private void Start()
+    {
+        _text = GetComponent<TMP_Text>();
+        _text.text = "Loading online count...";
+        StartCoroutine(UpdateOnlineCount());
+    }
+
+    private IEnumerator UpdateOnlineCount()
+    {
+        while (true)
+        {
+            using var uwr = UnityWebRequest.Get(GameShell.GetServerInterface().GetServerAddress() + "stats.json");
+            yield return uwr.SendWebRequest();
+
+            var stats = JsonConvert.DeserializeObject<ServerStats>(uwr.downloadHandler.text);
+            _text.text = $"{stats.OnlinePlayers} online | {stats.ActiveBattles} active battles";
+
+            yield return new WaitForSeconds(5);
+        }
+    }
+
+    public class ServerStats
+    {
+        public int OnlinePlayers { get; set; }
+        public int ActiveBattles { get; set; }
+    }
+}
\ No newline at end of file
diff --git a/Assets/Scripts/Assembly-CSharp/OnlineCountText.cs.meta b/Assets/Scripts/Assembly-CSharp/OnlineCountText.cs.meta
new file mode 100644
index 00000000..0f9b18a8
--- /dev/null
+++ b/Assets/Scripts/Assembly-CSharp/OnlineCountText.cs.meta
@@ -0,0 +1,3 @@
+fileFormatVersion: 2
+guid: 3ba3494398194c79a033a36cf20800a4
+timeCreated: 1777086225
\ No newline at end of file
diff --git a/Assets/prefabs/ui/entermatch/EnteringMatch.prefab b/Assets/prefabs/ui/entermatch/EnteringMatch.prefab
index 4d97cd63..d655bb26 100644
--- a/Assets/prefabs/ui/entermatch/EnteringMatch.prefab
+++ b/Assets/prefabs/ui/entermatch/EnteringMatch.prefab
@@ -7045,2 +7045,3 @@ MonoBehaviour:
   CancelMatchmakingButton: {fileID: 1161607090078389}
+  OnlineText: {fileID: 3629023196993208479}
   GameCamera: {fileID: 20694121525821480}
@@ -7451,2 +7452,3 @@ RectTransform:
   - {fileID: 224534704304255049}
+  - {fileID: 7152391952487535975}
   - {fileID: 224264357844037074}
@@ -7779 +7781,149 @@ MonoBehaviour:
   m_EditorClassIdentifier: 
+--- !u!1 &7730304590025242324
+GameObject:
+  m_ObjectHideFlags: 0
+  m_CorrespondingSourceObject: {fileID: 0}
+  m_PrefabInstance: {fileID: 0}
+  m_PrefabAsset: {fileID: 0}
+  serializedVersion: 6
+  m_Component:
+  - component: {fileID: 7152391952487535975}
+  - component: {fileID: 2416730669248565582}
+  - component: {fileID: 3629023196993208479}
+  - component: {fileID: 7003398451001004435}
+  m_Layer: 5
+  m_Name: OnlineText
+  m_TagString: Untagged
+  m_Icon: {fileID: 0}
+  m_NavMeshLayer: 0
+  m_StaticEditorFlags: 0
+  m_IsActive: 1
+--- !u!224 &7152391952487535975
+RectTransform:
+  m_ObjectHideFlags: 0
+  m_CorrespondingSourceObject: {fileID: 0}
+  m_PrefabInstance: {fileID: 0}
+  m_PrefabAsset: {fileID: 0}
+  m_GameObject: {fileID: 7730304590025242324}
+  m_LocalRotation: {x: 0, y: 0, z: 0, w: 1}
+  m_LocalPosition: {x: 0, y: 0, z: 0}
+  m_LocalScale: {x: 1, y: 1, z: 1}
+  m_ConstrainProportionsScale: 0
+  m_Children: []
+  m_Father: {fileID: 224291844617409189}
+  m_LocalEulerAnglesHint: {x: 0, y: 0, z: 0}
+  m_AnchorMin: {x: 0, y: 0}
+  m_AnchorMax: {x: 1, y: 0}
+  m_AnchoredPosition: {x: 0, y: 20}
+  m_SizeDelta: {x: -40, y: 100}
+  m_Pivot: {x: 0.5, y: 0}
+--- !u!222 &2416730669248565582
+CanvasRenderer:
+  m_ObjectHideFlags: 0
+  m_CorrespondingSourceObject: {fileID: 0}
+  m_PrefabInstance: {fileID: 0}
+  m_PrefabAsset: {fileID: 0}
+  m_GameObject: {fileID: 7730304590025242324}
+  m_CullTransparentMesh: 1
+--- !u!114 &3629023196993208479
+MonoBehaviour:
+  m_ObjectHideFlags: 0
+  m_CorrespondingSourceObject: {fileID: 0}
+  m_PrefabInstance: {fileID: 0}
+  m_PrefabAsset: {fileID: 0}
+  m_GameObject: {fileID: 7730304590025242324}
+  m_Enabled: 1
+  m_EditorHideFlags: 0
+  m_Script: {fileID: 11500000, guid: f4688fdb7df04437aeb418b961361dc5, type: 3}
+  m_Name: 
+  m_EditorClassIdentifier: 
+  m_Material: {fileID: 0}
+  m_Color: {r: 1, g: 1, b: 1, a: 1}
+  m_RaycastTarget: 1
+  m_RaycastPadding: {x: 0, y: 0, z: 0, w: 0}
+  m_Maskable: 1
+  m_OnCullStateChanged:
+    m_PersistentCalls:
+      m_Calls: []
+  m_text: Loading online count...
+  m_isRightToLeft: 0
+  m_fontAsset: {fileID: 11400000, guid: ce62afb04dbe15178e3ff84ff2726426, type: 2}
+  m_sharedMaterial: {fileID: -95363901605624429, guid: ce62afb04dbe15178e3ff84ff2726426,
+    type: 2}
+  m_fontSharedMaterials: []
+  m_fontMaterial: {fileID: 0}
+  m_fontMaterials: []
+  m_fontColor32:
+    serializedVersion: 2
+    rgba: 4278190080
+  m_fontColor: {r: 0, g: 0, b: 0, a: 1}
+  m_enableVertexGradient: 0
+  m_colorMode: 3
+  m_fontColorGradient:
+    topLeft: {r: 1, g: 1, b: 1, a: 1}
+    topRight: {r: 1, g: 1, b: 1, a: 1}
+    bottomLeft: {r: 1, g: 1, b: 1, a: 1}
+    bottomRight: {r: 1, g: 1, b: 1, a: 1}
+  m_fontColorGradientPreset: {fileID: 0}
+  m_spriteAsset: {fileID: 0}
+  m_tintAllSprites: 0
+  m_StyleSheet: {fileID: 0}
+  m_TextStyleHashCode: -1183493901
+  m_overrideHtmlColors: 0
+  m_faceColor:
+    serializedVersion: 2
+    rgba: 4294967295
+  m_fontSize: 100
+  m_fontSizeBase: 165
+  m_fontWeight: 400
+  m_enableAutoSizing: 1
+  m_fontSizeMin: 50
+  m_fontSizeMax: 200
+  m_fontStyle: 0
+  m_HorizontalAlignment: 2
+  m_VerticalAlignment: 4096
+  m_textAlignment: 65535
+  m_characterSpacing: 0
+  m_wordSpacing: 0
+  m_lineSpacing: 0
+  m_lineSpacingMax: 0
+  m_paragraphSpacing: 0
+  m_charWidthMaxAdj: 0
+  m_enableWordWrapping: 1
+  m_wordWrappingRatios: 0.4
+  m_overflowMode: 0
+  m_linkedTextComponent: {fileID: 0}
+  parentLinkedComponent: {fileID: 0}
+  m_enableKerning: 1
+  m_enableExtraPadding: 0
+  checkPaddingRequired: 0
+  m_isRichText: 1
+  m_parseCtrlCharacters: 1
+  m_isOrthographic: 1
+  m_isCullingEnabled: 0
+  m_horizontalMapping: 0
+  m_verticalMapping: 0
+  m_uvLineOffset: 0
+  m_geometrySortingOrder: 0
+  m_IsTextObjectScaleStatic: 0
+  m_VertexBufferAutoSizeReduction: 0
+  m_useMaxVisibleDescender: 1
+  m_pageToDisplay: 1
+  m_margin: {x: 0, y: 0, z: 0, w: 0}
+  m_isUsingLegacyAnimationComponent: 0
+  m_isVolumetricText: 0
+  m_hasFontAssetChanged: 0
+  m_baseMaterial: {fileID: 0}
+  m_maskOffset: {x: 0, y: 0, z: 0, w: 0}
+--- !u!114 &7003398451001004435
+MonoBehaviour:
+  m_ObjectHideFlags: 0
+  m_CorrespondingSourceObject: {fileID: 0}
+  m_PrefabInstance: {fileID: 0}
+  m_PrefabAsset: {fileID: 0}
+  m_GameObject: {fileID: 7730304590025242324}
+  m_Enabled: 1
+  m_EditorHideFlags: 0
+  m_Script: {fileID: 11500000, guid: 3ba3494398194c79a033a36cf20800a4, type: 3}
+  m_Name: 
+  m_EditorClassIdentifier: 
