From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nuutti <snuutti@protonmail.com>
Date: Wed, 29 Apr 2026 22:05:15 +0300
Subject: [PATCH] Add server alert system


diff --git a/Assets/Scripts/Assembly-CSharp/ChatManager.cs b/Assets/Scripts/Assembly-CSharp/ChatManager.cs
index 142b8783..313f1d26 100644
--- a/Assets/Scripts/Assembly-CSharp/ChatManager.cs
+++ b/Assets/Scripts/Assembly-CSharp/ChatManager.cs
@@ -140,2 +140,3 @@ public class ChatManager : SingletonMonoBehaviour<ChatManager>
 	{
+		onSystemMessageEvent += OnServerAlert;
 		if (_hasStarted)
@@ -154,2 +155,3 @@ public class ChatManager : SingletonMonoBehaviour<ChatManager>
 	{
+		onSystemMessageEvent -= OnServerAlert;
 		Disconnect();
@@ -167,2 +169,10 @@ public class ChatManager : SingletonMonoBehaviour<ChatManager>
 
+	private void OnServerAlert(string senderJid, string nickname, string subject, string message)
+	{
+		if (subject.Equals(":::ALERT"))
+		{
+			PopUpManager.ShowOk("Server Alert", message, null);
+		}
+	}
+
 	public void SetActiveChannel(ChannelType channel)
