From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nuutti <snuutti@protonmail.com>
Date: Fri, 27 Mar 2026 02:09:40 +0200
Subject: [PATCH] Fix MasterAudio music assignments


diff --git a/Assets/sound/MasterAudioContainer.prefab b/Assets/sound/MasterAudioContainer.prefab
index e373184d..4f65245f 100644
--- a/Assets/sound/MasterAudioContainer.prefab
+++ b/Assets/sound/MasterAudioContainer.prefab
@@ -22000,3 +22000,3 @@ MonoBehaviour:
       audLocation: 0
-      clip: {fileID: 8300000, guid: 4edc7a5e0c9f39ee8d6b70f3eb0c3049, type: 2}
+      clip: {fileID: 8300000, guid: 4edc7a5e0c9f39ee8d6b70f3eb0c3049, type: 3}
       songName: menu_music
@@ -22042,3 +22042,3 @@ MonoBehaviour:
       audLocation: 0
-      clip: {fileID: 8300000, guid: 1d35ffa69186e892c65372845878e023, type: 2}
+      clip: {fileID: 8300000, guid: 1d35ffa69186e892c65372845878e023, type: 3}
       songName: matchmaking2
@@ -22068,3 +22068,3 @@ MonoBehaviour:
       audLocation: 0
-      clip: {fileID: 8300000, guid: ff15b551cf3317af039628a99fe01e99, type: 2}
+      clip: {fileID: 8300000, guid: ff15b551cf3317af039628a99fe01e99, type: 3}
       songName: gameplay
@@ -22094,3 +22094,3 @@ MonoBehaviour:
       audLocation: 0
-      clip: {fileID: 8300000, guid: 9fe4455933ac4cab5f44090c7d29e94f, type: 2}
+      clip: {fileID: 8300000, guid: 9fe4455933ac4cab5f44090c7d29e94f, type: 3}
       songName: victory
